addons/pinniped/post-deploy/pkg/configure/concierge/concierge.go

```diff
@@ -73,3 +81,59 @@ func (c Configurator) CreateOrUpdateJWTAuthenticator(ctx context.Context, namesp
 	zap.S().Infof("Updated the JWTAuthenticator %s/%s", namespace, name)
 	return nil
 }
+
+// CreateOrUpdatePinnipedInfo creates pinniped information or updates existing data.
+func (c Configurator) CreateOrUpdatePinnipedInfo(ctx context.Context, pinnipedInfo supervisor.PinnipedInfo) error {
+	var err error
+	zap.S().Info("Creating the ConfigMap for Pinniped info")
+
+	// create configmap under kube-public namespace
+	pinnipedConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.PinnipedInfoConfigMapName,
+			Namespace: constants.KubePublicNamespace,
+		},
+		Data: map[string]string{
+			"pinniped_api_group_suffix":            pinnipedInfo.PinnipedAPIGroupSuffix,
```

Comment:
did we consider extracting this function and the analogous one in `supervisor.go` into a shared helper? seems like all but 3 lines are the same. we could probably get rid of ~50 lines of duplicate source code and ~200 lines of duplicate test code.

Reply:
I am very much pro doing this. Where should the helper function live?

Reply:
Made the change, lmk what you think!